node/hyperbahn/service_proxy.js

```diff
@@ -257,6 +262,85 @@ function updateExitNodes(exitNodes, svcchan) {
     }
 };
 
+ServiceDispatchHandler.prototype.isBlocked =
+function isBlocked(cn, serviceName) {
+    var self = this;
+    if (!self.blockingTable) {
+        return false;
+    }
+
+    cn = cn || '*';
+    serviceName = serviceName || '*';
+
+    if (self.blockingTable[cn] && (
+        self.blockingTable[cn][serviceName] ||
+        self.blockingTable[cn]['*'])) {
+        return true;
+    } else if (self.blockingTable['*'] && (
+        self.blockingTable['*'][serviceName] ||
+        self.blockingTable['*']['*'])) {
+        return true;
+    }
+
+    return false;
+};
+
+ServiceDispatchHandler.prototype.block =
+function block(cn, serviceName) {
+    var self = this;
+    cn = cn || '*';
```

Comment:
this interface is exposed through hyperbahn endpoint. At least one of cn/serviceName is required there.